TPP.Core/Commands/Definitions/BadgeCommands.cs

```diff
@@ -134,15 +141,150 @@ public async Task<CommandResult> SelectBadge(CommandContext context)
+
+                if( mode.Equals( PokedexModeMissing ))
+                {
+                    IEnumerable<PkmnSpecies> missingList = PokedexData.Load().KnownSpecies.Except( numBadgesPerSpecies.Keys );
+                    // Seems like PokedexData is not sorted. So we have to sort the list.
+                    IEnumerable<string> badgesFormatted = missingList.OrderBy( entry => entry.Id ).Select( entry => $"{entry}" );
+                    return new CommandResult
+                    {
+                        Response = isSelf
+                            ? $"You are currently missing the following badge(s): {string.Join(", ", badgesFormatted)}"
+                            : $"{user.Name} is currently missing the following badge(s): {string.Join(", ", badgesFormatted)}",
+                        ResponseTarget = ResponseTarget.WhisperIfLong
+                    };
+                }
+                else if( mode.Equals( PokedexModeDupes ) )
+                {
+                    var dupeList = numBadgesPerSpecies.Where(kvp => kvp.Value > 1);
+                    if( !dupeList.Any() )
+                    {
+                        return new CommandResult
+                        {
+                            Response = isSelf
+                                ? $"You do not own any duplicate Pokémon badges"
+                                : $"{user.Name} does not own any duplicate Pokémon badges"
+                        };
+                    }
+                    else
+                    {
+                        IEnumerable<string> badgesFormatted = dupeList.Select( kvp => $"{kvp.Key}" );
+                        return new CommandResult
+                        {
+                            Response = isSelf
+                                ? $"You are owning duplicates of the following badge(s): {string.Join(", ", badgesFormatted)}"
+                                : $"{user.Name} is owning duplicates of the following badge(s): {string.Join(", ", badgesFormatted)}",
+                            ResponseTarget = ResponseTarget.WhisperIfLong
+                        };
+                    }
+                }
+                else if( mode.Equals( PokedexModeComplementFromDupes ) )
+                {
+                    if( !optionalUser2.IsPresent )
+                    {
+                        return new CommandResult
+                        {
+                            Response = $"Mode {PokedexModeComplementFromDupes} requires a second user argument"
+                        };
+                    }
+                    else
+                    {
+                        var dupeList = numBadgesPerSpecies.Where( kvp => kvp.Value > 1 );
+                        ImmutableSortedDictionary<PkmnSpecies, int> myNumBadgesPerSpecies =
+                            await _badgeRepo.CountByUserPerSpecies(optionalUser2.Value.Id);
+
+                        var intersectList = dupeList.Except( myNumBadgesPerSpecies );
```

Comment:
looks like you implemented this opposite to how I expected it. That's fine, but the name "complement-from" is probably wrong then. I expected "A complement-from B" to print all badges B can provide for A, but it calculates the badges A can provide for B. Possibly just use "complements" instead of "complement-from" or something

Reply:
the term intersection is technically wrong here, since it's a difference. Think venn-diagram: the intersection is the thing in the middle, but the difference is one of the sides.

Reply:
imo using "my" in these variables like `myNumBadgesPerSpecies` is confusing since it doesn't necessarily reference the user issuing the command but any user. Something like `compareUser` might fit better

Reply:
dyschromatopsia and reading too much tutorials where people use variables starting with "my". I hate those too

Reply:
switching A and B was quite some trial and error, but it should work as intended now.